File: packages/rust/crates/oxbitnet-java/java/src/main/java/io/github/m96chan/oxbitnet/LoadProgress.java
package io.github.m96chan.oxbitnet;

/**
 * Immutable snapshot of model loading progress.
 *
 * @param phase    current phase ("download", "parse", or "upload")
 * @param loaded   bytes/items loaded so far
 * @param total    total bytes/items expected
 * @param fraction progress as a fraction (0.0 to 1.0)
 */
public record LoadProgress(String phase, long loaded, long total, double fraction) {
}

File: packages/rust/crates/oxbitnet-csharp/src/OxBitNet/LoadProgress.cs
namespace OxBitNet
{
    /// <summary>
    /// Phase of the model loading process.
    /// </summary>
    public enum LoadPhase
    {
        Download = 0,
        Parse = 1,
        Upload = 2,
    }

    /// <summary>
    /// Progress information during model loading.
    /// </summary>
    public sealed class LoadProgress
    {
        public LoadPhase Phase { get; }
        public ulong Loaded { get; }
        public ulong Total { get; }
        public double Fraction { get; }

        internal LoadProgress(LoadPhase phase, ulong loaded, ulong total, double fraction)
        {
            Phase = phase;
            Loaded = loaded;
            Total = total;
            Fraction = fraction;
        }
    }
}
